openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMModelCreator.java

```diff
@@ -104,18 +104,49 @@
     static final String ERROR_MSG_RANDOM_FOREST_REQUIRES_BAGGING =
             "Random Forest Boosting type requires bagging. Please see bagging parameters.";
 
+    /**
+     * Number of instances per C++ train data chunk.
+     * Train data is copied from the input stream into an array of chunks.
+     * Each chunk will have this many instances. Must be set before `fit()`.
+     * <p>
+     * Performance overhead notes:
+     * - Too small? Performance overhead - excessive in-memory data fragmentation.
+     * - Too large? RAM overhead - in the worst case the last chunk has only 1 instance.
+     *              Each instance might have upwards of 400 features. Each costs 8 bytes.
+     *              E.g.: 100k instances of 400 features =>  320MB / chunk
+     */
+    private final long trainDataChunkInstancesSize;
```

Comment:
I don't think the `LightGBMModelCreator` should be aware of this kind of implementation detail. IMO this belong to `LightGBMBinaryClassificationModelTrainer`